packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -0,0 +1,89 @@
+import { useContext, useEffect, useMemo, useState } from 'react';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+import { Post } from '../graphql/posts';
+import { FeedItem, PostItem } from './useFeed';
+
+interface UsePostModalNavigation {
+  onPrevious: () => void;
+  onNext: () => Promise<void>;
+  onOpenModal: (index: number) => void;
+  onCloseModal: () => void;
+  isFetchingNextPage?: boolean;
+  article: Post | null;
+}
+
+export const usePostModalNavigation = (
+  items: FeedItem[],
+  fetchPage: () => Promise<unknown>,
+): UsePostModalNavigation => {
+  const [openedPostIndex, setOpenedPostIndex] = useState<number>(null);
+  const [isFetchingNextPage, setIsFetchingNextPage] = useState(false);
+  const { trackEvent } = useContext(AnalyticsContext);
+
+  const ret = useMemo<UsePostModalNavigation>(
+    () => ({
+      isFetchingNextPage,
+      onCloseModal: () => setOpenedPostIndex(null),
+      onOpenModal: (index) => setOpenedPostIndex(index),
+      onPrevious() {
+        let index = openedPostIndex - 1;
+        for (; index > 0 && items[index].type !== 'post'; index -= 1);
+        const item = items[index];
+        if (item.type !== 'post') {
+          return;
+        }
+
+        trackEvent({
+          origin: 'article modal',
+          event_name: 'navigate previous',
+          target_id: item.post.id,
```

Comment:
This should be the old post (so the one we are coming from)
And we should actually use `postAnalyticsEvent` for all these, as it tracks post data.